aten/src/ATen/native/BinaryOps.cpp

```diff
@@ -224,14 +228,17 @@ template <typename Stub>
 Tensor& comparison_op_out(Tensor& result, const Tensor& self, const Tensor& other, Stub& stub) {
   TORCH_CHECK(result.scalar_type() == kBool,
               "The output tensor of lt must be a bool, but was ", result.scalar_type());
+  // Validate that is possible to convert zero-dim tensor's dtype to other dtype without overflow
```

Comment:
If they're both zero-dim, then TensorIterator will promote both types, right? So we shouldn't check for overflow when both inputs are zero-dim.

Reply:
Already changed this